### DISCUSSION TITLE
Open graph not working properly

### DISCUSSION THREAD

**Discussion by Masam-abbas-avialdo** [vercel/next.js#64053] [Help] [answered]
### Summary

I am getting open graph data in my browser, also in page source, but when i send link via any social media platform the open graph data does not display, i am trying to fetch data from api using async await and set that data in open graph. any suggestion please.
https://desibook.web.linkedunion.info/

**Comment by icyJoseph** [comment]
Hi,

I think there's an issue when rendering the app. When I use open graph testers, these fail to fetch the page.

Are you maybe rejecting bots or something, in middleware?

**Reply by Masam-abbas-avialdo** [reply]
No, I am not rejecting any bot. It was functioning properly a few days ago, but when I tested it recently, it did not work.

**Reply by icyJoseph** [reply]
For some reason, requesting your app with curl, does a redirect, I am guessing that for bots, for some reason it ends up in an infinite redirect loop.

**Reply by Masam-abbas-avialdo** [reply]
I am using fetch method to fetch data from api on server component.

**Reply by icyJoseph** [reply, marked as answer]
Could you describe your request flow? 

- Middleware? with redirects?
- Next.js redirects in next.config.js?
- Calls to `redirect` at render time?

As said, when requesting your page as a bot, it redirects with 307 endlessly

**Reply by Masam-abbas-avialdo** [reply]
The middleware only filter the domain and hit api according to that domain, if user is not authenticated he will be redirected to home page.

**Reply by icyJoseph** [reply]
I think that might be the problem? somehow it is redirecting the bots too... I also see it if I go in incognito mode:

<img width="1336" alt="Screenshot 2024-04-04 at 10 33 50" src="https://github.com/vercel/next.js/assets/21013447/c4bf960f-8bfa-4f6d-9330-74df3c4b36d6">

**Reply by Masam-abbas-avialdo** [reply]
Yes, I've just resolved it. There was an error in the middleware. Thank you, sir. Thank you very much.
